tests/cases/compiler/argumentsReferenceAndParameterWithInitializer1.ts

```diff
@@ -0,0 +1,131 @@
+// @strict: true
+// @checkJs: true
+// @lib: dom, esnext
+// @noEmit: true
+
+// @filename: index.js
+
+'use strict';
+
+// https://github.com/microsoft/TypeScript/issues/57435
+
+/** @type {globalThis['structuredClone']} */
+const structuredClone =
+  globalThis.structuredClone ??
+  function structuredClone (value, options = undefined) {
+    if (arguments.length === 0) {
+      throw new TypeError('missing argument')
+    }
+    return value;
+  }
+
+/** @type {(a: number, b: boolean | undefined, ...rest: string[]) => void} */
+const test1 = function(value, options = undefined) {
+  if (arguments.length === 0) {
```

Comment:
This is the only reference to `arguments` here. I think that perhaps this pattern (checking `arguments.length`) wouldn't have to make `containsArgumentsReference` to return `true`. I'm not sure if there is an appetite for this though.